modules/face/src/face_alignment.cpp

```diff
@@ -0,0 +1,906 @@
+    */
+    bool buildRegtree(regtree &tree,vector<training_sample>& samples,vector<Point2f> pixel_coordinates);
+    /** @brief This function greedily decides the best split among the test splits generated.
+    * @param samples vector storing the training samples
+    * @param pixel_coordinates vector for storing the random test coordinates generated..
+    * @param start stores start index of the samples currently at the node
+    * @param end stores end index of the samples currently at the node
+    */
+    splitr generateSplit(vector<Point2f> pixel_coordinates, vector<training_sample> samples,unsigned long start ,unsigned long end);
+    /**@brief This function randomly generates test coordinates for each level of cascade.
+    * @param pixel_coordinates stores the test coordinates generated for making splits.
+    * @param min_x minimum x coordinate of the random pixels generated
+    * @param max_x maximum x coordinate of the random pixels generated
+    * @param min_y minimum y coordinate of the random pixels generated
+    * @param max_y maximum y coordinate of the random pixels generated
+    */
+    void getTestCoordinates (vector< vector<Point2f> >& pixel_coordinates,float min_x,
+                                        float min_y, float max_x , float max_y);
+    /** @brief This function gets the landmarks in the meanshape nearest to the pixel coordinates.
+    * @param pixels stores the point nearest to which shape coorinates are to be determined.
+    */
+    unsigned long getNearestLandmark (Point2f pixels );
+    /** @brief This function implements gradient boosting by fitting regression trees
+    * to get weak learners and update the shape residual.
+    *@returns A forest of regression trees fitted which act as weak learners for the model.
+    *@param samples stores the training samples.
+    *@param pixel_coordinates stores the test pixel coordinates for making the split.
+    */
+    vector<regtree> gradientBoosting(vector<training_sample>& samples,vector<Point2f> pixel_coordinates);
+    /**@brief This function creates training sample by randomly assigning a current shape from set of
+    * shapes available.
+    *@returns a bool value
+    *@param samples An empty vector that would hold training samples generated.
+    *@param images vector which stores the images which are available as training data.
+    *@param landmarks vector which stores the actual shape of each corresponding image which is available as training data.
+    */
+    bool createTrainingSamples(vector<training_sample> &samples,vector<Mat> images,vector< vector<Point2f> > landmarks);
+};
+
+FacemarkKazemi::~FacemarkKazemi(){};
+
+bool FacemarkKazemiimpl:: getFaces(Mat src,vector<Rect> &facep,int select,double scaleFactor
+        , int minNeighbors, int flags, Size minSize, Size maxSize){
+    vector<Rect> facesp;
+    string face_cascade_name;
+    switch(select){
+        case FACE_HAAR : face_cascade_name = "haarcascade_frontalface_alt.xml" ;
```

Comment:
@sukhad-app 

__First.__ You should put your testdata to the _opencv_extra_ repository. Create folder _testdata/cv/face_ and put all your _txt_, _xml_ and _jpg_ files there (try to minimize the size of your testdata images, check _testdata/cv/shared_ folder and some others for reusable images like _lena.jpg_ etc.). Then in tests call function `cv::cvtest::findDataFile` like this:
  ```.cpp
std::string filename = cvtest::findDataFile("cv/face/1.txt", true); // true flag means - fail the test if the file not found
// do not forget to set OPENCV_TEST_DATA_PATH environment variable 
// to the location of your opencv_extra/testdata directory
```
Then publish it in your _opencv_extra_ GitHub fork on branch _face_alignment_ and create a PR to the upstream repository (it is important to have branch name same as in _opencv_contrib_ PR).

----------

__Second.__ Modify `getFaces` method to accept `CascadeClassifier` instance:
```.cpp
bool FacemarkKazemiImpl:: getFaces(Mat src, vector<Rect> &facep, Ptr<cv::CascadeClassifier> face_cascade, 
   double scaleFactor, int minNeighbors, int flags, Size minSize, Size maxSize)
{
   // ...
```
or store CascadeClassifier instance inside FacemarkKaszemiImpl object:
```.cpp
FacemarkKazemiImpl::FacemarkKazemiImpl(Ptr<cv::CascadeClassifier> face_cascade_)
    : face_cascade(face_cascade_)
{
    // ...